bothan-core/src/ipfs/client.rs

```diff
@@ -0,0 +1,33 @@
+use reqwest::{Client, StatusCode};
+
+use crate::ipfs::errors::Error;
+
+pub struct IpfsClient {
+    url: String,
+    client: Client,
+}
+
+impl IpfsClient {
+    pub fn new(url: String, client: Client) -> Self {
+        Self { url, client }
+    }
+
+    pub async fn get_ipfs<T: AsRef<str>>(&self, hash: T) -> Result<String, Error> {
+        let url = format!("{}/ipfs/{}", self.url, hash.as_ref());
```

Comment:
it's part of the ipfs spec, I thinking leaving it out of the url is viable in case we decide to use different paths. [ref](https://specs.ipfs.tech/http-gateways/path-gateway/)